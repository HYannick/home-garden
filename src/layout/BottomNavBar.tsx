/** @jsx jsx */
import React, {useEffect, useState, Fragment, MouseEventHandler} from 'react';
import {jsx} from "@emotion/core";
import styled from "@emotion/styled";
import Drop from "../core/svg/Drop";
import {NavLink, RouteComponentProps, withRouter} from "react-router-dom";
import Add from "../core/svg/Add";
import Profile from '../core/svg/Profile';
import {Overlay} from "./Header";
import plant from '../core/svg/green-tea.svg';

interface BottomNavBarProps {
  isOpen?: boolean
}

interface NavOverlayProps extends BottomNavBarProps {
  onClick: MouseEventHandler,
}

const IconWrapper = styled('div')<BottomNavBarProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 6rem;
  cursor: pointer;
  & svg {
    width: 2.5rem;
    height: 2.5rem;
  }
  transform: rotate(${({isOpen}) => isOpen ? '45deg' : 0});
  transition: transform 0.3s;
`;

const NavWrapper = styled('div')`
  height: 7rem;
  border-radius: 5rem 5rem 0 0;
  box-shadow: 0 -0.3rem 0.6rem 0rem rgba(196, 196, 196, 0.15);
  background-color: ${({theme}) => theme.palette.light};
  display: flex;
  align-items: center;
  justify-content: space-between; 
  & > * {
   flex: 1;
  }
  & > .nav__link {
    position: relative;
    z-index: 1;
    transition: opacity 0.3s;
    &:before {
      content : '';
      position: absolute;
      z-index: -1;
      background-color: ${({theme}) => theme.palette.primary.light};
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%) scale(0.6);
      opacity: 0;
      width: 5rem;
      height: 5rem;
      border-radius: 5rem;
      transition: opacity 0.3s , transform 0.3s ;
    }
    &:after {
      content : '';
      position: absolute;
      left: 50%;
      bottom: 0.8rem;
      opacity: 0;
      z-index: 1;
      width: 0.6rem;
      height: 0.6rem;
      border-radius: 5rem;
      background-color: ${({theme}) => theme.palette.primary.dark};
      transform: translateX(-50%);
      transition: opacity 0.3s , transform 0.3s ;
    }
  }
  & > .nav__link--active {
    &:before {
      transform: translate(-50%, -50%) scale(1);
      opacity: 1;
    }
    &:after {
      opacity: 1;
    }
    & .outlined {
      & svg path {
        fill: none;
        stroke: ${({theme}) => theme.palette.primary.dark};
      }
    }
    & svg path {
      fill: ${({theme}) => theme.palette.primary.dark};
      stroke: ${({theme}) => theme.palette.primary.dark};
      transition: fill 0.3s , stroke 0.3s;
    }
  }
`;

const OptionsWrapper = styled('div')`
  background-color: ${({theme}) => theme.palette.light};
  & ul {
    margin: 0 auto;
    padding: 0 2rem 1rem;
    max-width: 30rem;
  }
  
  & li {
    list-style: none;
    text-align: center;
  }
  
  a {
    color:  ${({theme}) => theme.palette.grey.dark};
    font-weight: 600;
    font-size: 2rem;
    text-decoration: none;
    display: block;
    padding: 1rem;
  }
`;

const MainWrapper = styled('div')<BottomNavBarProps>`
  position: fixed;
  z-index: 1;
  bottom: 0;
  width: 100%;
  transition: transform 0.3s;
  transform: translateY(${({isOpen}) => isOpen ? 0 : '12.3rem'});
  .nav__link {
    opacity: ${({isOpen}) => isOpen ? 0 : 1};
    transition: opacity 0.3s visibility 0.3s;
    pointer-events:  ${({isOpen}) => isOpen ? 'none' : 'auto'}; 
  }
`;


const AddWrapper = styled('div')<BottomNavBarProps>`
  position: relative;
  z-index: 1;
  & svg path {
    fill: ${({isOpen, theme}) => isOpen && theme.palette.danger.dark};
    transition: fill 0.3s , stroke 0.3s;
  }
  &:before {
      content : '';
      position: absolute;
      z-index: -1;
      background-color: ${({theme}) => theme.palette.danger.light};
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%) scale(${({isOpen}) => isOpen ? 1 : 0.4});
      opacity: ${({isOpen}) => isOpen ? 1 : 0};
      width: 5rem;
      height: 5rem;
      border-radius: 5rem;
      transition: opacity 0.3s , transform 0.3s ;
  }
`;

const NavOverlay = styled(Overlay)<NavOverlayProps>`
  z-index: 0; 
  background-color:  ${({theme}) => theme.palette.grey.darker};
  opacity: ${({isOpen}) => isOpen ? 0.8 : 0};
  pointer-events: ${({isOpen}) => isOpen ? 'auto' : 'none'};
  transition: opacity 0.3s;
`;

const Divider = styled('div')`
  width: 100%;
  height: 0.1rem;
  margin: 1rem 0;
  background-color:  ${({theme}) => theme.palette.grey.light};
`;

const Infos = styled('div')`
  position: relative;
  padding: 2rem;
  text-align: center;
  margin-bottom: 1rem;
    &:before {
    content: url(${plant});
    width: 8rem;
    height: 8rem;
    position: absolute;
    bottom: -1rem;
    right: 2rem;
  }
  h2 {
    color: ${({theme}) => theme.palette.primary.light};
    font-size: 4rem;
  }
  p {
    color: ${({theme}) => theme.palette.light};
    font-size: 1.5rem;
  }
  
  &:after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translateX(-50%);
    width: 10rem;
    height: 0.6rem;
    background-color: ${({theme}) => theme.palette.light};
    border-radius: 2rem;
  }
`;


export const BottomNavBar: React.FC<RouteComponentProps> = ({location}) => {
  const [isOpen, expand] = useState(false);

  const hide = () => expand(false);
  const toggle = () => expand(!isOpen);

  useEffect(() => {
    hide()
  }, [location]);

  if (location.pathname.match('/onboarding')) {
    return null;
  }

  if (location.pathname.match('/create')) {
    return null;
  }

  const iconDefaultColor = '#707070';

  return (
    <Fragment>
      <MainWrapper isOpen={isOpen}>
        <div>
          {
            isOpen && (
              <Infos onClick={hide}>
                <h2>Want to make the family grow?</h2>
                <p>Take a photo of your plant<br/>or search it in our database</p>
              </Infos>
            )
          }
          <NavWrapper>
            <NavLink to="/" exact activeClassName="nav__link--active" className="nav__link">
              <IconWrapper className="outlined">
                <Drop stroke={iconDefaultColor} fill="none"/>
              </IconWrapper>
            </NavLink>
            <AddWrapper isOpen={isOpen}>
              <IconWrapper data-testid="add" onClick={toggle} isOpen={isOpen}>
                <Add fill={iconDefaultColor}/>
              </IconWrapper>
            </AddWrapper>
            <NavLink to="/profile" exact activeClassName="nav__link--active" className="nav__link">
              <IconWrapper>
                <Profile fill={iconDefaultColor} stroke={iconDefaultColor}/>
              </IconWrapper>
            </NavLink>
          </NavWrapper>
        </div>
        <OptionsWrapper>
          <ul>
            <li><NavLink to="/create">Take a photo</NavLink></li>
            <li><Divider/></li>
            <li><NavLink to="/search">Search in library</NavLink></li>
          </ul>
        </OptionsWrapper>
      </MainWrapper>
      <NavOverlay onClick={hide} isOpen={isOpen}/>
    </Fragment>
  );
};

export default withRouter(BottomNavBar);
