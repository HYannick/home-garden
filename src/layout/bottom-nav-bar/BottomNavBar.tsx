/** @jsx jsx */
import React, { useEffect, useState, Fragment } from 'react';
import { jsx } from '@emotion/core';
import { NavLink, RouteComponentProps, withRouter } from 'react-router-dom';
import Drop from '../../core/svg/Drop';
import Add from '../../core/svg/Add';
import Profile from '../../core/svg/Profile';
import {
  // AddWrapper,
  Divider,
  IconWrapper,
  Infos,
  MainWrapper,
  NavOverlay,
  NavWrapper,
  OptionsWrapper,
} from './BottomNavBar.styled';


export const BottomNavBar: React.FC<RouteComponentProps> = ({ location }) => {
  const [isOpen, expand] = useState(false);

  const hide = () => expand(false);
  // const toggle = () => expand(!isOpen);

  useEffect(() => {
    hide();
  }, [location]);

  if (
    location.pathname.match('/onboarding') ||
    location.pathname.match('/create') ||
    location.pathname.match('/plants')
  ) {
    return null;
  }

  const iconDefaultColor = '#707070';

  return (
    <Fragment>
      <MainWrapper isOpen={isOpen}>
        <div>
          {
            isOpen && (
              <Infos onClick={hide}>
                <h2>Want to make the family grow?</h2>
                <p>Take a photo of your plant<br/>or search it in our database</p>
              </Infos>
            )
          }
          <NavWrapper>
            <NavLink to="/" exact activeClassName="nav__link--active" className="nav__link">
              <IconWrapper className="outlined">
                <Drop stroke={iconDefaultColor} fill="none"/>
              </IconWrapper>
            </NavLink>
            <NavLink to="/search" exact activeClassName="nav__link--active" className="nav__link no-stroke">
              <IconWrapper data-testid="add">
                <Add fill={iconDefaultColor}/>
              </IconWrapper>
            </NavLink>
            {/* <AddWrapper isOpen={isOpen}> */}
            {/*  <IconWrapper data-testid="add" onClick={toggle} isOpen={isOpen}> */}
            {/*    <Add fill={iconDefaultColor}/> */}
            {/*  </IconWrapper> */}
            {/* </AddWrapper> */}
            <NavLink to="/profile" exact activeClassName="nav__link--active" className="nav__link">
              <IconWrapper>
                <Profile fill={iconDefaultColor} stroke={iconDefaultColor}/>
              </IconWrapper>
            </NavLink>
          </NavWrapper>
        </div>
        <OptionsWrapper>
          <ul>
            <li><NavLink to="/create">Add your plant</NavLink></li>
            <li><Divider/></li>
            <li><NavLink to="/search">Search in library</NavLink></li>
          </ul>
        </OptionsWrapper>
      </MainWrapper>
      <NavOverlay onClick={hide} isOpen={isOpen}/>
    </Fragment>
  );
};

export default withRouter(BottomNavBar);
