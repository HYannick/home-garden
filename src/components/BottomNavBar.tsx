/** @jsx jsx */
import React, {useEffect, useState, Fragment} from 'react';
import {jsx} from "@emotion/core";
import styled from "@emotion/styled";
import Drop from "../core/svg/Drop";
import {NavLink, RouteComponentProps, withRouter} from "react-router-dom";
import Add from "../core/svg/Add";
import Profile from '../core/svg/Profile';
import {Overlay} from "../layout/Header";
import plant from '../core/svg/green-tea.svg'

const IconWrapper = styled('div')`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 6rem;
  cursor: pointer;
  & svg {
    width: 2.5rem;
    height: 2.5rem;
  }
  transform: rotate(${(props: any) => props.isOpen ? '45deg' : 0});
  transition: transform 0.3s;
`;

const NavWrapper = styled('div')`
  height: 7rem;
  border-radius: 5rem 5rem 0 0;
  box-shadow: 0 -0.3rem 0.6rem 0.6rem rgba(196, 196, 196, 0.15);
  background-color: white;
  display: flex;
  align-items: center;
  justify-content: space-between; 
  & > * {
   flex: 1;
  }
  & > .nav__link {
    position: relative;
    z-index: 1;
    transition: opacity 0.3s;
    &:before {
      content : '';
      position: absolute;
      z-index: -1;
      background-color: ${(props: any) => props.theme.palette.primary.light};
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%) scale(0.6);
      opacity: 0;
      width: 5rem;
      height: 5rem;
      border-radius: 5rem;
      transition: opacity 0.3s , transform 0.3s ;
    }
    &:after {
      content : '';
      position: absolute;
      left: 50%;
      bottom: 0.8rem;
      opacity: 0;
      z-index: 1;
      width: 0.6rem;
      height: 0.6rem;
      border-radius: 5rem;
      background-color: ${(props: any) => props.theme.palette.primary.dark};
      transform: translateX(-50%);
      transition: opacity 0.3s , transform 0.3s ;
    }
  }
  & > .nav__link--active {
    &:before {
      transform: translate(-50%, -50%) scale(1);
      opacity: 1;
    }
    &:after {
      opacity: 1;
    }
    & .outlined {
      & svg path {
        fill: none;
        stroke: ${(props: any) => props.theme.palette.primary.dark};
      }
    }
    & svg path {
      fill: ${(props: any) => props.theme.palette.primary.dark};
      stroke: ${(props: any) => props.theme.palette.primary.dark};
      transition: fill 0.3s , stroke 0.3s;
    }
  }
`;

const OptionsWrapper = styled('div')`
  background-color: white;
  & ul {
    margin: 0 auto;
    padding: 0 2rem 1rem;
    max-width: 30rem;
  }
  
  & li {
    list-style: none;
    text-align: center;
  }
  
  a {
    color:  ${(props: any) => props.theme.palette.grey.dark};
    font-weight: 600;
    font-size: 2rem;
    text-decoration: none;
  }
`;

const MainWrapper = styled('div')`
  position: fixed;
  z-index: 1;
  bottom: 0;
  width: 100%;
  transition: transform 0.3s;
  transform: translateY(${(props: any) => props.isOpen ? 0 : '8.3rem'});
  .nav__link {
    opacity: ${(props: any) => props.isOpen ? 0 : 1};
    transition: opacity 0.3s visibility 0.3s;
    pointer-events:  ${(props: any) => props.isOpen ? 'none' : 'auto'}; 
  }
`;

const AddWrapper = styled('div')`
  position: relative;
  z-index: 1;
  & svg path {
    fill: ${(props: any) => props.isOpen && props.theme.palette.danger.dark};
    transition: fill 0.3s , stroke 0.3s;
  }
  &:before {
      content : '';
      position: absolute;
      z-index: -1;
      background-color: ${(props: any) => props.theme.palette.danger.light};
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%) scale(${(props: any) => props.isOpen ? 1 : 0.4});
      opacity: ${(props: any) => props.isOpen ? 1 : 0};
      width: 5rem;
      height: 5rem;
      border-radius: 5rem;
      transition: opacity 0.3s , transform 0.3s ;
  }
`;

const NavOverlay = styled(Overlay)`
  z-index: 0; 
  background-color:  ${(props: any) => props.theme.palette.grey.darker};
  opacity: ${(props: any) => props.isOpen ? 0.8 : 0};
  pointer-events: ${(props: any) => props.isOpen ? 'auto' : 'none'};
  transition: opacity 0.3s;
`;

const Divider = styled('div')`
  width: 100%;
  height: 0.1rem;
  margin: 1rem 0;
  background-color:  ${(props: any) => props.theme.palette.grey.light};
`;

const Infos = styled('div')`
  position: relative;
  padding: 2rem;
  text-align: center;
  margin-bottom: 1rem;
    &:before {
    content: url(${plant});
    width: 8rem;
    height: 8rem;
    position: absolute;
    bottom: -1rem;
    right: 2rem;
  }
  h2 {
    color: ${(props: any) => props.theme.palette.primary.light};
    font-size: 4rem;
  }
  p {
    color: white;
    font-size: 1.5rem;
  }
  
  &:after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translateX(-50%);
    width: 10rem;
    height: 0.6rem;
    background-color: white;
    border-radius: 20px;
  }
`;
const BottomNavBar: React.FC<RouteComponentProps> = ({location}) => {
  const [isOpen, expand] = useState(false);

  const hide = () => expand(false);
  const toggle = () => expand(!isOpen);

  useEffect(() => {
    hide()
  }, [location]);

  if (location.pathname.match('/onboarding')) {
    return null;
  }

  const title = '';

  return (
    <Fragment>
      <MainWrapper {...{isOpen}}>
        <div>
          {
            isOpen && (
              <Infos onClick={hide}>
                <h2>Want to make the family grow?</h2>
                <p>Take a photo of your plant<br/>or search it in our database</p>
              </Infos>
            )
          }
          <NavWrapper>
            <NavLink to="/" exact activeClassName="nav__link--active" className="nav__link">
              <IconWrapper className="outlined">
                <Drop stroke="#707070" fill="none"/>
              </IconWrapper>
            </NavLink>
            <AddWrapper {...{isOpen}}>
              <IconWrapper onClick={toggle} {...{isOpen}}>
                <Add fill="#707070"/>
              </IconWrapper>
            </AddWrapper>
            <NavLink to="/profile" exact activeClassName="nav__link--active" className="nav__link">
              <IconWrapper>
                <Profile fill="#707070" stroke="#707070"/>
              </IconWrapper>
            </NavLink>
          </NavWrapper>
        </div>
        <OptionsWrapper>
          <ul>
            <li><NavLink to="/photo">Take a photo</NavLink></li>
            <li>
              <Divider/>
            </li>
            <li><NavLink to="/search">Search in library</NavLink></li>
          </ul>
        </OptionsWrapper>
      </MainWrapper>
      <NavOverlay onClick={hide} {...{title, isOpen}}/>
    </Fragment>
  );
};

export default withRouter(BottomNavBar);
