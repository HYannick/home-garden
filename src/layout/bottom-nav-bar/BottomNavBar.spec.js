import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderWithTheme } from '../../theme-wrapper';
import { BottomNavBar } from './BottomNavBar';

afterEach(cleanup);
describe('BottomNavBar', () => {
  const props = {
    location: { pathname: '' },
    match: jest.fn(),
    history: {},
  };
  it('should match snapshot on closed', () => {
    const { container } = renderWithTheme(
      <Router><BottomNavBar {...props}/></Router>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot on open', () => {
    const { getByTestId, container } = renderWithTheme(
      <Router><BottomNavBar {...props}/></Router>,
    );
    const button = getByTestId('add');
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});
