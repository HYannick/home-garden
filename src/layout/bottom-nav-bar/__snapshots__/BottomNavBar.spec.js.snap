// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavBar should match snapshot on closed 1`] = `
<div>
  <div
    class="css-lrv8g5"
  >
    <div>
      <div
        class="css-drx1bz"
      >
        <a
          aria-current="page"
          class="nav__link nav__link--active"
          href="/"
        >
          <div
            class="outlined css-1xi8t4l"
          >
            <svg
              viewBox="0 0 14.214 17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.944 1s6.27 6.2 6.27 9.329A5.9 5.9 0 0 1 7.107 16 5.9 5.9 0 0 1 1 10.329C1 7.193 6.944 1 6.944 1z"
                fill="none"
                stroke="#707070"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </a>
        <div
          class="css-o0luf0"
        >
          <div
            class="css-1xi8t4l"
            data-testid="add"
          >
            <svg
              viewBox="0 0 14.999 14.999"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.328 13.828V8.672H1.172a1.172 1.172 0 0 1 0-2.344h5.156V1.172a1.172 1.172 0 0 1 2.344 0v5.156h5.156a1.172 1.172 0 0 1 0 2.344H8.672v5.156a1.172 1.172 0 0 1-2.344 0z"
                fill="#707070"
              />
            </svg>
          </div>
        </div>
        <a
          class="nav__link"
          href="/profile"
        >
          <div
            class="css-1xi8t4l"
          >
            <svg
              viewBox="0 0 15.286 21"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.643 9.071a4.286 4.286 0 1 0-4.286-4.285 4.286 4.286 0 0 0 4.286 4.286zm0-7.143a2.857 2.857 0 1 1-2.857 2.857 2.857 2.857 0 0 1 2.857-2.856zm0 8.571A7.143 7.143 0 0 0 .5 17.643v2.143a.715.715 0 0 0 1.429 0v-2.143a5.715 5.715 0 0 1 11.429 0v2.143a.715.715 0 0 0 1.429 0v-2.143A7.143 7.143 0 0 0 7.643 10.5z"
                fill="#707070"
                stroke="#707070"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
    <div
      class="css-s56s06"
    >
      <ul>
        <li>
          <a
            href="/create"
          >
            Take a photo
          </a>
        </li>
        <li>
          <div
            class="css-rcl9j8"
          />
        </li>
        <li>
          <a
            href="/search"
          >
            Search in library
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="css-ajruyc"
  />
</div>
`;

exports[`BottomNavBar should match snapshot on open 1`] = `
<div>
  <div
    class="css-1lamfzg"
  >
    <div>
      <div
        class="css-1hqf3h3"
      >
        <h2>
          Want to make the family grow?
        </h2>
        <p>
          Take a photo of your plant
          <br />
          or search it in our database
        </p>
      </div>
      <div
        class="css-drx1bz"
      >
        <a
          aria-current="page"
          class="nav__link nav__link--active"
          href="/"
        >
          <div
            class="outlined css-1xi8t4l"
          >
            <svg
              viewBox="0 0 14.214 17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.944 1s6.27 6.2 6.27 9.329A5.9 5.9 0 0 1 7.107 16 5.9 5.9 0 0 1 1 10.329C1 7.193 6.944 1 6.944 1z"
                fill="none"
                stroke="#707070"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </a>
        <div
          class="css-qr370g"
        >
          <div
            class="css-989axa"
            data-testid="add"
          >
            <svg
              viewBox="0 0 14.999 14.999"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.328 13.828V8.672H1.172a1.172 1.172 0 0 1 0-2.344h5.156V1.172a1.172 1.172 0 0 1 2.344 0v5.156h5.156a1.172 1.172 0 0 1 0 2.344H8.672v5.156a1.172 1.172 0 0 1-2.344 0z"
                fill="#707070"
              />
            </svg>
          </div>
        </div>
        <a
          class="nav__link"
          href="/profile"
        >
          <div
            class="css-1xi8t4l"
          >
            <svg
              viewBox="0 0 15.286 21"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.643 9.071a4.286 4.286 0 1 0-4.286-4.285 4.286 4.286 0 0 0 4.286 4.286zm0-7.143a2.857 2.857 0 1 1-2.857 2.857 2.857 2.857 0 0 1 2.857-2.856zm0 8.571A7.143 7.143 0 0 0 .5 17.643v2.143a.715.715 0 0 0 1.429 0v-2.143a5.715 5.715 0 0 1 11.429 0v2.143a.715.715 0 0 0 1.429 0v-2.143A7.143 7.143 0 0 0 7.643 10.5z"
                fill="#707070"
                stroke="#707070"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
    <div
      class="css-s56s06"
    >
      <ul>
        <li>
          <a
            href="/create"
          >
            Take a photo
          </a>
        </li>
        <li>
          <div
            class="css-rcl9j8"
          />
        </li>
        <li>
          <a
            href="/search"
          >
            Search in library
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="css-1obqsrn"
  />
</div>
`;
